pymc3/sampling.py

```diff
@@ -1150,13 +1205,13 @@ def sample_posterior_predictive(trace,
         if progressbar:
             indices.close()
 
+    ppc_trace = ppc_trace_t.trace_dict
     if keep_size:
         for k, ary in ppc_trace.items():
-            ary = np.asarray(ary)
             ppc_trace[k] = ary.reshape((nchain, len_trace, *ary.shape[1:]))
         return ppc_trace
```

Comment:
I agree with @ColCarroll here, we should just dedent and `return ppc_trace` here.

Reply:
Done